runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/MonitoringUtil.java

```diff
@@ -219,6 +220,11 @@ public static String getGcloudCancelCommand(DataflowPipelineOptions options, Str
   }
 
   public static State toState(String stateName) {
-    return MoreObjects.firstNonNull(DATAFLOW_STATE_TO_JOB_STATE.get(stateName), State.UNRECOGNIZED);
+    @Nullable State recognizedState = DATAFLOW_STATE_TO_JOB_STATE.get(stateName);
+    if (recognizedState != null) {
+      return recognizedState;
+    }
+    LOG.warn("Unrecognized state from Dataflow service: {}", stateName);
```

Comment:
Should we add a note: "Consider upgrading to a newer version of Beam" and/or an explanation: "This likely means that you are using an old version of Beam SDK". ?

Reply:
Yea good idea. I will update the message. FWIW we could refactor things to somehow get the string out (unfortunately enums are not helpful for attaching it) but since it is pretty rare it isn't worth too much effort yet.